packages/service-backend/src/section/section.service.ts

```diff
@@ -126,42 +153,131 @@ export class SectionService {
+            title: lesson.title,
+            status: userLesson?.status || Status.NOT_STARTED,
+            numberOfQuestions: questionsCount || 0,
+            numberOfLectures: lecturesCount || 0,
+          };
+        },
+      );
 
-        const questionsCount = await prisma.question.count({
-          where: {
-            lesson_id: lesson.id,
+      return await Promise.all(lessonPromises);
+    } catch (error) {
+      Logger.error(error);
+      throw new NotFoundException('Lessons not found !');
+    }
+  }
+
+  async joinSection(
+    sectionId: string,
+    userId: string,
+  ): Promise<SectionIdResponse> {
+    try {
+      let userToSection = await prisma.usertoSection.findUnique({
+        where: {
+          section_id_user_id: {
+            user_id: userId,
+            section_id: sectionId,
           },
-        });
+        },
+      });
 
-        const lecturesCount = await prisma.lecture.count({
-          where: {
-            lesson_id: lesson.id,
+      if (!userToSection) {
+        userToSection = await prisma.usertoSection.create({
+          data: {
+            user_id: userId,
+            section_id: sectionId,
+            status: Status.IN_PROGRESS,
           },
         });
+      }
 
-        return {
-          id: lesson.id,
-          description: lesson.description,
-          title: lesson.title,
-          status: userLesson?.status || LessonStatus.NOT_STARTED,
-          numberOfQuestions: questionsCount || 0,
-          numberOfLectures: lecturesCount || 0,
-        };
-      });
-
-      return await Promise.all(lessonPromises);
+      if (!userToSection) {
+        Logger.error('Failed to create user section !');
+        throw new NotFoundException('Failed to create user section !');
+      }
+      return { id: userToSection.section_id } as SectionIdResponse;
     } catch (error) {
       Logger.error(error);
-      throw new NotFoundException('Lessons not found !');
+      throw new ConflictException('User Section not created !');
+    }
+  }
+
+  static async UpdateSectionCompletionFromLesson(
+    lessonId: string,
+    userId: string,
+  ) {
+    const sectionId = await prisma.lesson.findUnique({
+      where: { id: lessonId },
+      select: {
+        section_id: true,
+      },
+    });
+
+    const sectionLessonsIds = (
+      await prisma.lesson.findMany({
+        where: {
+          section_id: sectionId.section_id,
+        },
+        select: { id: true },
+      })
+    ).map((lesson) => lesson.id);
+
+    const sectionUserToLessonStatuses: Status[] = (
+      await prisma.usertoLesson.findMany({
+        where: {
+          id: { in: sectionLessonsIds },
+        },
+        select: { status: true },
+      })
+    ).map((userToLessonObject) => userToLessonObject.status);
+
+    if (
+      sectionUserToLessonStatuses.every(
+        (status) => status === Status.COMPLETED,
+      ) === true &&
+      sectionUserToLessonStatuses.length === sectionLessonsIds.length
+    ) {
```

Comment:
We could replace all this complex code by a simple sql query like this
```
select count(*)
from "Lesson" as L
where section_id = '<section_id>'
and not exists(
    select 1
    from  "UsertoLesson" utl
    where utl.lesson_id = L.id
        and utl.status != 'COMPLETED'
);
```

Let see how to do that in Prisma.

Reply:
```
prisma.lesson.count({
      where: {
        UsertoLesson: {
          none: {
            status: {
              not: 'COMPLETED'
            }
          }
        }
      }
    })
```